pages/index.js

```diff
@@ -35,7 +35,7 @@ export default function Home() {
             <ul className="flex items-center">
               <li>
                 <BsFillMoonStarsFill
-                  onClick={() => setDarkMode(!darkMode)}
+                  onClick={() => setDarkMode(prev => !prev)}
```

Comment:
This was just for optimization, more of best practices
Since setDarkMode happens asynchronously, other updates could happen and change the value of darkMode, if you are going to calculate the state based on the previous state, you have to make sure you have the last and most up to date value,